src/mapml/layers/MapLayer.js

```diff
@@ -386,6 +497,172 @@ export var MapMLLayer = L.Layer.extend({
+        extent.onmousedown = (downEvent) => {
+          if(downEvent.target.tagName.toLowerCase() === "input" || downEvent.target.tagName.toLowerCase() === "select") return;
+          downEvent.preventDefault();
+          downEvent.stopPropagation();
+
+          let control = extent,
+              controls = extent.parentNode,
+              moving = false, yPos = downEvent.clientY;
+
+              document.body.onmousemove = (moveEvent) => {
+                moveEvent.preventDefault();
+    
+                // Fixes flickering by only moving element when there is enough space
+                let offset = moveEvent.clientY - yPos;
+                moving = Math.abs(offset) > 5 || moving;
+                if( (controls && !moving) || (controls && controls.childElementCount <= 1) || 
+                    controls.getBoundingClientRect().top > control.getBoundingClientRect().bottom || 
+                    controls.getBoundingClientRect().bottom < control.getBoundingClientRect().top){
+                      return;
+                    }
+                
+                controls.classList.add("mapml-draggable");
+                control.style.transform = "translateY("+ offset +"px)";
+                control.style.pointerEvents = "none";
+    
+                let x = moveEvent.clientX, y = moveEvent.clientY,
+                    root = mapEl.tagName === "MAPML-VIEWER" ? mapEl.shadowRoot : mapEl.querySelector(".mapml-web-map").shadowRoot,
+                    elementAt = root.elementFromPoint(x, y),
+                    swapControl = !elementAt || !elementAt.closest("fieldset") ? control : elementAt.closest("fieldset");
+          
+                swapControl =  Math.abs(offset) <= swapControl.offsetHeight ? control : swapControl;
+                
+                control.setAttribute("aria-grabbed", 'true');
+                control.setAttribute("aria-dropeffect", "move");
+                if(swapControl && controls === swapControl.parentNode){
+                  swapControl = swapControl !== control.nextSibling? swapControl : swapControl.nextSibling;
+                  if(control !== swapControl){ 
+                    yPos = moveEvent.clientY;
+                    control.style.transform = null;
+                  }
+                  controls.insertBefore(control, swapControl);
+                }
+              };
+
+              document.body.onmouseup = () => {
+                control.setAttribute("aria-grabbed", "false");
+                control.removeAttribute("aria-dropeffect");
+                control.style.pointerEvents = null;
+                control.style.transform = null;
+                let controlsElems = controls.children,
+                    zIndex = 0;
+                for(let c of controlsElems){
+                  let extentEl = c.querySelector("span").extent;
+                  
+                  extentEl.setAttribute("data-moving","");
+                  mapEl.insertAdjacentElement("beforeend", extentEl);
```

Comment:
When you drag an extent to re-order it, the \<map-extent> elements (all) get re-parented into the mapml-viewer element.